src/researches/getPassiveVoice.js

```diff
@@ -1,110 +1,122 @@
-			break;
-		case "en":
-		default:
-			sentenceParts = getSentencePartsDefault( sentence, "en" );
-			break;
-	}
-	return sentenceParts;
+let getPassivesMorphological = function( sentenceObjects, language ) {
+	let passiveSentencesByType = [];
+
+	forEach( sentenceObjects, function( sentence ) {
+		let strippedSentence = stripHTMLTags( sentence.getSentenceText() ).toLocaleLowerCase();
+
+		sentence.setPassive( determinePassiveSentence( strippedSentence, language ) );
+
+		if ( sentence.isPassive() === true ) {
+			passiveSentencesByType.push( sentence.getSentenceText() );
+		}
+	} );
+	return {
+		passiveSentencesByType,
+	};
 };
 
 /**
- * Checks the sentence part for any passive verb.
+ * Looks for periphrastic passive voice.
  *
- * @param {object} sentencePart The sentence part object to check for passives.
- * @param {string} language The language to use for finding passive verbs.
- * @returns {boolean} True if passive is found, false if no passive is found.
+ * @param {Array} sentenceObjects Sentences extracted from the text.
+ * @param {string} language Language of the text.
+ * @returns {object} The number of passives found in the text and the passive sentences.
  */
-var determinePassives = function( sentencePart, language ) {
-	switch( language ) {
-		case "de":
-			sentencePart.setPassive( determinePassivesGerman( sentencePart.getSentencePartText(), sentencePart.getAuxiliaries() ), "de" );
-			break;
-		case "fr":
-			sentencePart.setPassive( determinePassivesDefault( sentencePart.getSentencePartText(), sentencePart.getAuxiliaries(), "fr" ) );
-			break;
-		case "es":
-			sentencePart.setPassive( determinePassivesDefault( sentencePart.getSentencePartText(), sentencePart.getAuxiliaries(), "es" ) );
-			break;
-		case "en":
-		default:
-			sentencePart.setPassive( determinePassivesDefault( sentencePart.getSentencePartText(), sentencePart.getAuxiliaries(), "en" ) );
-			break;
+let getPassivesPeriphrastic = function( sentenceObjects, language ) {
+	let passiveSentencesByType = [];
+	forEach( sentenceObjects, function( sentence ) {
+		let strippedSentence = stripHTMLTags( sentence.getSentenceText() ).toLocaleLowerCase();
+		let sentenceParts = [];
+
+		if ( language === "de" ) {
+			sentenceParts = getSentencePartsGerman( strippedSentence );
+		} else {
+			sentenceParts = getSentencePartsDefault( strippedSentence, language );
+		}
+
+		let passive = false;
+		forEach( sentenceParts, function( sentencePart ) {
+			sentencePart.setPassive( determinePassiveSentencePart( sentencePart.getSentencePartText(), sentencePart.getAuxiliaries(), language ) );
+			passive = passive || sentencePart.isPassive();
+		} );
+		if ( passive === true ) {
+			passiveSentencesByType.push( sentence.getSentenceText() );
+		}
+	} );
+	return {
+		passiveSentencesByType,
+	};
+};
+
+/**
+ * Determines which passive voice marking should be executed for a text based on the language.
+ *
+ * @param {Array} sentenceObjects Sentences extracted from the text.
```

Comment:
Please review the whole file and change `sentenceObjects` into `sentences` as per the comment above:

> The name of this can just be sentences. Try to avoid the type of variables in the name of variables. I know this hasn't been done consistently in the project, but that is legacy.

Reply:
Agreed, fixed.